src/asset/MountArchive.cpp

```diff
@@ -0,0 +1,89 @@
+#include "MountArchive.h"
+
+#include "bit7z/bitarchivereader.hpp"
+#include "utils/Log.h"
+
+#include <cstddef>
+#include <filesystem>
+#include <format>
+#include <string>
+#include <vector>
+
+MountArchive::MountArchive(std::filesystem::path archive_path)
+    : Mount(archive_path), reader(new bit7z::BitArchiveReader(library, path.string())) {
+}
+
+MountArchive::~MountArchive() {
+    MountArchive::save_cache();
+}
+
+Mount::MountType MountArchive::get_type() {
+    return ARCHIVE;

```

Comment:
Forcing derived classes to return this enum value defined in the base class breaks the open-closed principle. C++'s type and polymorphism system is more advanced than this and you can make static inferences about this stuff at compile time. Needing to do this is a code smell.